### ISSUE TITLE
[Request] Distance based travel Time between worlds

### ISSUE THREAD

**Issue by StargateMC** [zmaster587/AdvancedRocketry#1784]
**What do you think is missing or what do you think can be done better**

At the moment players can travel between worlds almost instantly... or you can increase the duration between worlds universally.

I'm proposing we implement a feature whereby travel time is based on the distance between worlds instead.

I've actually implemented this in a custom build of AR and keen to know if it would be accepted as a PR.

**Comment by zmaster587** [comment]
There should already be variation in time between planet calculated by the distance between them.  There's is a max time.  Maybe the difference needs to be made more extreme?

**Comment by StargateMC** [comment]
I believe you are thinking of your warp fuel costing?

Time to travel appears to be calcualted here:

https://github.com/zmaster587/AdvancedRocketry/blob/master/src/main/java/zmaster587/advancedRocketry/stations/SpaceObjectManager.java#L409

If It is something that is already in, then it definately needs to be made more extreme.. we couldnt tell the difference between travelling a distance of 1000 between stars compared to 10.

**Comment by zmaster587** [comment]
No, I'm sure I'm thinking of time, it's calculated here https://github.com/zmaster587/AdvancedRocketry/blob/1.12/src/main/java/zmaster587/advancedRocketry/tile/station/TileWarpShipMonitor.java#L455 and passed to `moveStationToBody` as the variable `timeDelta`.

Since fuel cost is already a function of distance and there's only one warp core tier (it'll need to be decoupled later), time is calculated as a function of fuel consumption to reduce code redundancy.

If you want to change the how long warp takes, that's probably where you want to do it.

**Comment by StargateMC** [comment]
I think that the reason this appears to be the same for interstellar trips is here:

https://github.com/zmaster587/AdvancedRocketry/blob/1.12/src/main/java/zmaster587/advancedRocketry/tile/station/TileWarpShipMonitor.java#L102

Perhaps we can adjust to factor in the distance between stars instead?

**Comment by StargateMC** [comment]
Id suggest a calculation of:

1. Allowing anyone to get anywhere in 1 jump:

fuelCost = (WARP_CAPACITY / GALAXY_MAX_SIZE) * DISTANCE;

or:

2. Allowing people to range limit ships - this would need an additional config:

fuelCost = (CONF_MAX_FUEL_COST/GALAXY_MAX_SIZE) * DISTANCE;

That being said I'm sure you can come up with a better way to do this.
If you were adding a config, could you also add one for the warp ship for capacity as well please? 

Happy to do a PR if you're keen on something like that?

**Comment by zmaster587** [comment]
Its alright, its a lsrge code base.  I wouldn't be opposed to having it factor in the distance between stars for timing calculation.

I do have a concern about making the time it takes too long though.  Not everyone using the mod will have large stations with something to do on them while in warp.  In fact, it's likely that lower teir travel setups will be used by smaller stations since they're more likely to be in earlyish game.

Alternatively, if long times are allowed with the default config, there needs to be a way to tell the player how long warp will take since it cannot be canceled.

For WARP_CAPACITY and GALAXY_MAX_SIZE, where would these, values cone from?

**Comment by StargateMC** [comment]
Hi @zmaster587 

Galaxy max size would be the maximum size of the planet selector in theory.

Warp fuel capacity would come from the warp controller. 

Your right an ETA is definately needed.
Id suggest it be shown on planets in the selectors as well as time remaining on current warp be shown on the warp controller.

In addition to this I'd suggest adding a backtrack option... so a player can effectively undo their current warp by travelling back to their previous location, taking the same amount of time as theyve already spent.

It might be a good idea to add new config options for warp ship fuel capacity, max ship range (for a single jump) etc.

**Comment by StargateMC** [comment]
> I do have a concern about making the time it takes too long though. Not everyone using the mod will have large stations with something to do on them while in warp. In fact, it's likely that lower teir travel setups will be used by smaller stations since they're more likely to be in earlyish game.

Perhaps we can have a maximum range for each warp? Or rely on server owners to set a warp speed multiplier like you are already?
